### ISSUE TITLE
Visual submissions may be rotated differently on list versus detail pages

### ISSUE THREAD

**Issue by huntertur** [Weasyl/weasyl#1436]
Today, while browsing through art on the production website, I discovered that visual submissions may sometimes be rotated differently on a user's submission list page compared to the submission's detail page.

# Example I noticed

* User page with rotated submissions (*Bath dragon*, *Aurora dragon*, *Vulcano dragon*): https://www.weasyl.com/submissions/rivalmit
* Detail pages with correct rotation:
  * https://www.weasyl.com/~rivalmit/submissions/2394225/bath-dragon
  * https://www.weasyl.com/~rivalmit/submissions/2394224/aurora-dragon
  * https://www.weasyl.com/~rivalmit/submissions/2394223/vulcano-dragon

# Other information

* It looks like the full-size source images (which display correctly) are JPEGs, while the incorrectly-rotated images are WebP thumbnails.
* The incorrect rotation is present on multiple browsers. Tested browsers:
  * Mozilla/5.0 (X11; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0
  * Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0
  * Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36

**Comment by huntertur** [comment]
One possible lead on this: it looks like rotation is specified via Exif metadata. The incorrect rotation seen on parts of the website matches the "original" orientation.

![image](https://github.com/user-attachments/assets/b3dace73-cca1-41cb-964e-fd6a901cab17)

**Comment by charmander** [comment]
Yeah, the Pillow thumbnailing has this and a color profile issue, while the ImageMagick-based resizing works correctly. (Unfortunately, our use of ImageMagick is based on a [fragile and unmaintained package](https://github.com/Weasyl/sanpera), so it’s not a matter of just using it instead.) Will publish the replacement for both soon.

**Comment by huntertur** [comment]
Sounds good! If there is anything I can do to assist, please let me know and I will be happy to help.

I'm curious, is the upcoming replacement Pillow-based or ImageMagick-based, or something new entirely?
